### PR TITLE
Infra changes required for stable builds

### PR DESCRIPTION
**Pull request by Anipik**
- nowarn is required for a shipping package to depend on non-shipping package for core-setup
- adding back CopyNupkgAndChangeVersion.cs bcoz this is only used in stable versions
- making Microsoft.Extensions.HostFactoryResolver, System.Numerics.Tensors & Internal.Transport as non-shipping, non-stable

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -86,6 +86,7 @@
 
     <!-- We don't want Private packages to be shipped to NuGet.org -->
     <IsShippingPackage Condition="($(MSBuildProjectName.Contains('Private')) or $(IsExperimentalAssembly)) and '$(MSBuildProjectExtension)' == '.pkgproj'">false</IsShippingPackage>
+    <StabilizePackageVersion Condition="'$(IsShippingPackage)' == 'false'">false</StabilizePackageVersion>
```
Comment (safern) [inline Directory.Build.props:89]: I thought this was implicitly done by arcade?

File: src/libraries/pkg/Microsoft.Extensions.Internal.Transport/Microsoft.Extensions.Internal.Transport.pkgproj [hunk 2]
```diff
@@ -5,6 +5,8 @@
     <SkipValidatePackage>true</SkipValidatePackage>
     <HarvestStablePackage>false</HarvestStablePackage>
     <IsShipping>false</IsShipping>
+    <StabilizePackageVersion>false</StabilizePackageVersion>
+    <StableVersion></StableVersion>
```
Comment (safern) [inline src/libraries/pkg/Microsoft.Extensions.Internal.Transport/Microsoft.Extensions.Internal.Transport.pkgproj:9]: I thought this was handled by arcade when a package is marked as non-shipping.

File: src/libraries/libraries-packages.proj [hunk 3]
```diff
@@ -44,6 +44,9 @@
       -->
       <PkgProjects Include="$(PkgDir)*\*.pkgproj" Exclude="$(MSBuildThisFileDirectory)pkg\*Private*\*.pkgproj" />
       <PkgProjects Include="*\pkg\**\*.pkgproj" />
+      <PkgProjects Remove="Microsoft.Extensions.HostFactoryResolver\pkg\Microsoft.Extensions.HostFactoryResolver.Sources.pkgproj" />
```
Comment (safern) [inline src/libraries/libraries-packages.proj:47]: Why bother doing all the other changes if these packages are note going to be built at all in servicing branch?

File: tools-local/tasks/installer.tasks/CopyNupkgAndChangeVersion.cs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+using NuGet.Versioning;
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class CopyNupkgAndChangeVersion : BuildTask
```
Comment (safern) [inline tools-local/tasks/installer.tasks/CopyNupkgAndChangeVersion.cs:15]: Why is this needed? I don't see you including any usage of it in your PR?

Comment (Anipik) [reply tools-local/tasks/installer.tasks/CopyNupkgAndChangeVersion.cs:15]: The usage is already checked in https://github.com/dotnet/runtime/blob/release/5.0/src/installer/tests/PrepareTestAssets/PrepareTestAssets.proj#L129
We are not hitting this error in master because this target is only executed while producing stable bits.

File: src/libraries/libraries-packages.proj (hunk 3 above)
Comment (Anipik) [reply src/libraries/libraries-packages.proj:47]: https://github.com/dotnet/runtime/pull/42872/files/0b646c327798f236ce6664f74cecfd4208d39898#diff-db6238b87f150b0d817e43057d645249R45 
is excluding all the private packages from stable, without excluding these they end up being stable non-shipping packages.

**Comment by ViktorHofer** [comment]
> nowarn is required for a shipping package to depend on non-shipping package for core-setup

what is the non-shipping package here?

File: src/libraries/Directory.Build.targets [hunk 5]
```diff
@@ -274,6 +274,7 @@
 
   <PropertyGroup>
     <SkipLocalsInit Condition="'$(SkipLocalsInit)' == '' and '$(MSBuildProjectExtension)' == '.csproj' and '$(IsNETCoreAppSrc)' == 'true' and '$(TargetFramework)' == '$(NetCoreAppCurrent)'">true</SkipLocalsInit>
+    <VersionSuffix>$(_PreReleaseLabel)$(_BuildNumberLabels)</VersionSuffix>
```
Comment (ViktorHofer) [inline src/libraries/Directory.Build.targets:277]: Usually we set the VersionSuffix in Versions.props. Why here only for libraries?

**Review by ViktorHofer — APPROVED** [review]
LGTM but please follow-up on the questions before merging.

File: src/libraries/Directory.Build.targets (hunk 5 above)
Comment (Anipik) [reply src/libraries/Directory.Build.targets:277]: VersionSuffix should be empty when stablizePackageVersion is set. We require this because of this library specific error check. https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Build.Tasks.Packaging/src/build/Packaging.targets#L706
We have the same thing done in 3.1 as well. cc @ericstj 
Not sure why this error check is here, we should probably resolve this for 6.0

File: src/installer/pkg/projects/netcoreapp/sfx/Microsoft.NETCore.App.SharedFx.sfxproj [hunk 6]
```diff
@@ -2,6 +2,7 @@
   <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
 
   <PropertyGroup>
+    <NoWarn>$(NoWarn);NU1603</NoWarn>
```
Comment (ericstj) [inline src/installer/pkg/projects/netcoreapp/sfx/Microsoft.NETCore.App.SharedFx.sfxproj:5]: We should not suppress this.  That warning indicates a newer version of a package was restored rather than the one specified.  We should fix the version.

File: src/libraries/libraries-packages.proj (hunk 3 above)
Comment (safern) [reply src/libraries/libraries-packages.proj:47]: Yeah, that is what I mean, since we're not building these packages at all on servicing branch, why bother doing all the other changes in their `Directory.Build.props`, etc?

Comment (safern) [reply src/libraries/libraries-packages.proj:47]: But I just saw that you reverted those changes.

File: src/installer/pkg/projects/netcoreapp/sfx/Microsoft.NETCore.App.SharedFx.sfxproj (hunk 6 above)
Comment (Anipik) [reply src/installer/pkg/projects/netcoreapp/sfx/Microsoft.NETCore.App.SharedFx.sfxproj]: i checked and we can remove this now.

File: src/libraries/Directory.Build.targets (hunk 5 above)
Comment (ViktorHofer) [reply src/libraries/Directory.Build.targets:277]: So should we follow-up on this?

Comment (ericstj) [reply src/libraries/Directory.Build.targets:277]: https://github.com/dotnet/arcade/issues/6305
